Rotate CustomDrop Model

DomScripts opened the pull request:
The goal of this PR is aimed at adding the ability to rotate the entity spawned when using `CustomDrop()`. This was done by adding a rotation to the entity along with a coordinate offset. The rotation and coordinate offset must be done after `PlaceObjectOnGroundProperly()` as it seems to reset the entity's rotation after calling in testing calling the native after applying the offsets. 

Using the native `SetEntityRotation()` requires a rotationOrder. 5 was chosen as it is the standard XYZ but different rotational orders can be found [here](https://docs.fivem.net/natives/?_0xAFBD61CC738D9EB9) if a different one is prefered.

Handling the case if the user doesn't want to set a rotation and offset was also considered by giving a default vec3(0,0,0) in it's place which would make the entity spawned use it's default values. This was chosen to be done when the point is created instead of the time when the entity is created so the entity's offset and rotation can be stored in the points data if someone would ever need that information.

Hakkodevelopment commented:
Seems useless, ngl

thelindat commented:
What's the point of an offset when you're already setting the coords of the drop?

DomScripts commented:
The point of an offset is to properly set whatever model you are using on the ground properly. During testing, if you apply the rotation to the entity then call `PlaceObjectOnGroundProperly()` it will reset the rotation of the entity in order to place it on the ground correctly. Therefore you need to apply the rotation after calling the native. The issue you run into then is that if you rotate the entity after setting the coordinates it will no longer be on the ground properly. 

This image below showcases the behavior if you apply a rotation after calling `PlaceObjectOnGroundProperly()` without an offset. The model is no longer on the ground correctly after apply a -90.0 rotation in the X-axis.
![image](https://github.com/user-attachments/assets/58a1a9dc-439a-4029-8bbf-857bd5e3b862)

This image below shows the behavior if you apply the rotation after calling `PlaceObjectOnGroundProperly()` with an offset to the model. 
![image](https://github.com/user-attachments/assets/c6a7cb42-882e-49b1-9ee6-718461b87430)

I'm not familiar with any other native that places the object on the ground correctly while keeping the rotation of the object. Just using the native `PlaceObjectOnGroundProperly()` isn't enough in order to achieve this so my approach was to add a coordinate offset to the model in order for people to be able to lineup the models correctly with the ground.


When it comes to the default rotation and offset values. I have removed them and replaced them with checks if they are defined instead in order to not have the useless objects.
